driver/custom_endpoint_monitor.cc

```diff
@@ -47,55 +46,81 @@ AWS_SDK_HELPER SDK_HELPER;
                                                  const std::string& endpoint_identifier, const std::string& region,
-                                                 DataSource* ds, bool enable_logging)
-  : custom_endpoint_host_info(custom_endpoint_host_info),
-    endpoint_identifier(endpoint_identifier),
-    region(region),
-    enable_logging(enable_logging) {
+                                                 long long refresh_rate_nanos, ctpl::thread_pool& thread_pool,
+                                                 bool enable_logging)
+    : topology_service(topology_service),
+      custom_endpoint_host(custom_endpoint_host),
+      endpoint_identifier(endpoint_identifier),
+      region(region),
+      refresh_rate_nanos(refresh_rate_nanos),
+      thread_pool(thread_pool),
+      enable_logging(enable_logging) {
   if (enable_logging) {
     this->logger = init_log_file();
   }
 
-  ++SDK_HELPER;
+  this->run();
+}
 
-  Aws::RDS::RDSClientConfiguration client_config;
-  if (!region.empty()) {
-    client_config.region = region;
+#ifdef UNIT_TEST_BUILD
+CUSTOM_ENDPOINT_MONITOR::CUSTOM_ENDPOINT_MONITOR(const std::shared_ptr<TOPOLOGY_SERVICE> topology_service,
+                                                 const std::string& custom_endpoint_host,
+                                                 const std::string& endpoint_identifier, const std::string& region,
+                                                 long long refresh_rate_nanos, ctpl::thread_pool& thread_pool,
+                                                 bool enable_logging, std::shared_ptr<Aws::RDS::RDSClient> client)
+    : topology_service(topology_service),
+      custom_endpoint_host(custom_endpoint_host),
+      endpoint_identifier(endpoint_identifier),
+      region(region),
+      refresh_rate_nanos(refresh_rate_nanos),
+      thread_pool(thread_pool),
+      enable_logging(enable_logging) {
+  if (enable_logging) {
+    this->logger = init_log_file();
   }
-
-  this->rds_client = std::make_shared<Aws::RDS::RDSClient>(
-    Aws::Auth::DefaultAWSCredentialsProviderChain().GetAWSCredentials(), client_config);
-
   this->run();
 }
+#endif
 
 bool CUSTOM_ENDPOINT_MONITOR::should_dispose() { return true; }
 
 bool CUSTOM_ENDPOINT_MONITOR::has_custom_endpoint_info() const {
   auto default_val = std::shared_ptr<CUSTOM_ENDPOINT_INFO>(nullptr);
-  return custom_endpoint_cache.get(this->custom_endpoint_host_info->get_host(), default_val) != default_val;
+  return custom_endpoint_cache.get(this->custom_endpoint_host, default_val) != default_val;
 }
 
 void CUSTOM_ENDPOINT_MONITOR::run() {
-  this->thread_pool.resize(1);
-  this->thread_pool.push([=](int id) {
-    MYLOG_TRACE(this->logger, 0, "Starting custom endpoint monitor for '%s'",
-                this->custom_endpoint_host_info->get_host().c_str());
+  MYLOG_TRACE(this->logger, 0, "Starting custom endpoint monitor for '%s'", this->custom_endpoint_host.c_str());
```

Comment:
It looks like a monitor only starts if `thread_pool.size()` is not `1`, So, should this log message go after the `if` check below?